File: impacted-test-engine/src/main/java/com/teamscale/test_impacted/engine/ImpactedTestEngineConfiguration.java
package com.teamscale.test_impacted.engine;

import com.teamscale.test_impacted.engine.executor.ITestSorter;
import com.teamscale.test_impacted.engine.executor.TeamscaleAgentNotifier;
import org.junit.platform.engine.TestEngine;

/** Container for a configuration used by the {@link ImpactedTestEngine} */
public class ImpactedTestEngineConfiguration {

	/** The directory to write testwise coverage and available tests to. */
	final TestDataWriter testDataWriter;

	/** The test engine registry used to determine the {@link TestEngine}s to use. */
	final TestEngineRegistry testEngineRegistry;

	/** The {@link ITestSorter} to use for execution of tests. */
	final ITestSorter testSorter;

	/** An API to signal test start and end to the agent. */
	final TeamscaleAgentNotifier teamscaleAgentNotifier;

	public ImpactedTestEngineConfiguration(
			TestDataWriter testDataWriter,
			TestEngineRegistry testEngineRegistry,
			ITestSorter testSorter, TeamscaleAgentNotifier teamscaleAgentNotifier ) {
		this.testDataWriter = testDataWriter;
		this.testEngineRegistry = testEngineRegistry;
		this.testSorter = testSorter;
		this.teamscaleAgentNotifier = teamscaleAgentNotifier;
	}
}

File: impacted-test-engine/src/main/kotlin/com/teamscale/test_impacted/engine/ImpactedTestEngineConfiguration.kt
package com.teamscale.test_impacted.engine

import com.teamscale.test_impacted.engine.executor.ITestSorter
import com.teamscale.test_impacted.engine.executor.TeamscaleAgentNotifier

/**
 * Configuration class for the Impacted Test Engine, responsible for managing dependencies and behavior
 * required for executing impacted tests. It provides access to utilities for test data management,
 * test engine discovery, test sorting, and interaction with external agents.
 *
 * @property testDataWriter Used to handle the writing of test execution data and test details to files.
 * @property testEngineRegistry Manages and provides access to available test engines, filtering and iterating over them.
 * @property testSorter Defines the logic for selecting and sorting tests for execution.
 * @property teamscaleAgentNotifier Facilitates communication with the Teamscale test-wise coverage agent by signaling test lifecycle events.
 */
class ImpactedTestEngineConfiguration(
	val testDataWriter: TestDataWriter,
	val testEngineRegistry: TestEngineRegistry,
	val testSorter: ITestSorter,
	val teamscaleAgentNotifier: TeamscaleAgentNotifier
)